transit_odp/timetables/utils.py

```diff
@@ -217,13 +217,33 @@ def get_holidays_records_to_insert(records):
             )
 
 
+def get_filtered_rows_by_journeys(
+    df: pd.DataFrame, journey_mappings: dict
+) -> pd.DataFrame:
+    return df[
+        df.apply(lambda row: filter_rows_by_journeys(row, journey_mappings), axis=1)
+    ]
+
+
+def get_journey_mappings(df: pd.DataFrame) -> dict:
+    return (
+        df.groupby(["file_id", "vehicle_journey_code"])["day_of_week"]
+        .unique()
+        .apply(list)
+        .to_dict()
+    )
+
+
 def filter_rows_by_journeys(row, journey_mapping):
     date_obj = row["exceptions_date"]
     if date_obj:
```

Comment:
This will fail when `row["exceptions_operational"]` is False - when the exceptions operation is False we want to return `day_of_week in operational_days`